packages/cli/src/tools/packageManager.js

```diff
@@ -61,8 +67,34 @@ export function uninstall(packageNames: Array<string>, options?: Options) {
     : executeCommand('npm', ['uninstall', ...packageNames, '--save'], options);
 }
 
-export function installAll(options?: Options) {
-  return shouldUseYarn(options)
-    ? executeCommand('yarn', ['install'], options)
-    : executeCommand('npm', ['install'], options);
+export function installCocoaPods() {
+  return executeCommand('sudo', ['gem', 'install', 'cocoapods'], {
+    silent: false,
+  });
+}
+
+export async function installPods(options?: Options) {
+  try {
+    await commandExists('pod');
+  } catch (err) {
+    return {error: ERRORS.noCocoaPods};
+  }
+
+  process.chdir('ios');
+
+  try {
+    return await executeCommand('pod', ['install'], options);
+  } catch (err) {
+    return {error: ERRORS.failedToRunPodInstall};
+  }
+}
+
+export async function installAll(options?: Options) {
+  shouldUseYarn(options)
+    ? await executeCommand('yarn', ['install'], options)
+    : await executeCommand('npm', ['install'], options);
+
+  // TODO: only do this on macOS
+
+  return installPods(options);
```

Comment:
Not sure if it makes sense to leave pods inside of `PackageManager` (since it's currently yarn and npm) but it's a process to install dependencies anyway.

Reply:
Personally, I would leave `PackageManager` as is (since it's also used in different places I guess) and create a different function, `installPods`. 

That way, you will not have to throw your "special" errors, you could just handle the logic inside the `catch` block, e.g.:
```
 let isPodsInstalled = false;
 try {
    await commandExists('pod');
    isPodsInstalled = true;
  } catch (err) {
    isPodsInstalled = await installCocoaPods(); // assuming you can skip it
  }

  if (isPodsInstalled) { installCocoaPodsDependencies() } else {}
```

Reply:
This is obviously a pseudocode so I'd improve it!

Reply:
Alright, gonna work with this in mind.